node/test/hyperbahn/rate-limiting.js

```diff
@@ -0,0 +1,260 @@
+                logger: cluster.logger,
+                bob: bob,
+                steve: steve,
+                assert: assert
+            };
+            series([
+                send.bind(null, opts),
+                send.bind(null, opts),
+                function sendError(done) {
+                    var tchannelJSON = TChannelJSON({
+                        logger: cluster.logger
+                    });
+                    tchannelJSON.send(bob.clientChannel.request({
+                        timeout: 5000,
+                        serviceName: steve.serviceName
+                    }), 'echo', null, 'hello', function onResponse(err, res) {
+
+                        assert.ok(err && err.type === 'tchannel.busy' &&
+                            err.message === 'steve is rate limited by 2 qps',
+                            'should be rate limited');
+                        done();
+                    });
+                }
+            ], function done() {
+                steveHyperbahnClient.destroy();
+                assert.end();
+            });
+        }
+    });
+
+    HyperbahnCluster.test('total rate limiting works', {
+        size: 1,
+        timers: timers,
+        kValue: 1,
+        rateLimitingBuckets: 2,
+        totalQpsLimit: 2
+    }, function t(cluster, assert) {
+        var steve = cluster.remotes.steve;
+        var bob = cluster.remotes.bob;
+
+        var steveHyperbahnClient = new HyperbahnClient({
+            serviceName: steve.serviceName,
+            callerName: 'forward-test',
+            hostPortList: cluster.hostPortList,
+            tchannel: steve.channel,
+            logger: DebugLogtron('hyperbahnClient')
+        });
+        steveHyperbahnClient.once('advertised', onAdvertised);
+        steveHyperbahnClient.advertise();
+
+        function onAdvertised() {
+            var opts = {
+                logger: cluster.logger,
+                bob: bob,
+                steve: steve,
+                assert: assert
+            };
+            series([
+                send.bind(null, opts),
+                wait,
+                function sendError(done) {
+                    var tchannelJSON = TChannelJSON({
+                        logger: cluster.logger
+                    });
+                    tchannelJSON.send(bob.clientChannel.request({
+                        timeout: 5000,
+                        serviceName: steve.serviceName
+                    }), 'echo', null, 'hello', function onResponse(err, res) {
+                        assert.ok(err && err.type === 'tchannel.busy' &&
+                            err.message === 'hyperbahn node is rate limited by the total rate 2 qps',
+                            'should be rate limited');
+                        done();
+                    });
+                }
+            ], function done() {
+                steveHyperbahnClient.destroy();
+                assert.end();
+            });
+        }
+    });
```

Comment:
Perhaps add a negative test to verify that the rate limiter does not engage when (a) there is smooth and steady load just beneath maxRequestsPerMs (b) there is bursty load just beneath maxRequestsPerMs on average over the window.